parsl/executors/globus_compute.py

```diff
@@ -0,0 +1,171 @@
+    Refer to `globus-compute user documentation <https://globus-compute.readthedocs.io/en/latest/executor.html>`_
+    and `reference documentation <https://globus-compute.readthedocs.io/en/latest/reference/executor.html>`_
+    for more details.
+
+    .. note::
+       As a remote execution system, Globus Compute relies on serialization to ship
+       tasks and results between the Parsl client side and the remote Globus Compute
+       Endpoint side. Serialization is unreliable across python versions, and
+       wrappers used by Parsl assume identical Parsl versions across on both sides.
+       We recommend using matching Python, Parsl and Globus Compute version on both
+       the client side and the endpoint side for stable behavior.
+
+    """
+
+    @typeguard.typechecked
+    def __init__(
+            self,
+            endpoint_id: UUID_LIKE_T,
+            task_group_id: Optional[UUID_LIKE_T] = None,
+            resource_specification: Optional[Dict[str, Any]] = None,
+            user_endpoint_config: Optional[Dict[str, Any]] = None,
+            label: str = "GlobusComputeExecutor",
+            batch_size: int = 128,
+            amqp_port: Optional[int] = None,
+            client: Optional[Client] = None,
+            **kwargs,
+    ):
+        """
+        Parameters
+        ----------
+
+        endpoint_id:
+            id of the endpoint to which to submit tasks
+
+        task_group_id:
+            The Task Group to which to associate tasks.  If not set,
+            one will be instantiated.
+
+        resource_specification:
+            Specify resource requirements for individual task execution.
+
+        user_endpoint_config:
+            User endpoint configuration values as described
+            and allowed by endpoint administrators. Must be a JSON-serializable dict
+            or None. Refer docs from `globus-compute
+            <https://globus-compute.readthedocs.io/en/latest/endpoints/endpoints.html#templating-endpoint-configuration>`_
+            for more info.
+
+        label:
+            a label to name the executor
+
+        batch_size:
+            the maximum number of tasks to coalesce before
+            sending upstream [min: 1, default: 128]
+
+        amqp_port:
+            Port to use when connecting to results queue. Note that the
+            Compute web services only support 5671, 5672, and 443.
+
+        client:
+            instance of globus_compute_sdk.Client to be used by the executor.
+            If not provided, the executor will instantiate one with default arguments.
+
+        kwargs:
+            Other kwargs listed will be passed through to globus_compute_sdk.Executor
+            as is. Refer to `globus-compute docs
+            <https://globus-compute.readthedocs.io/en/latest/reference/executor.html#globus-compute-executor>`_
+        """
+        super().__init__()
+        self.endpoint_id = endpoint_id
+        self.task_group_id = task_group_id
```

Comment:
There's some part of me that looks at this and says " 'just' pass in a Globus Compute `executor` as the single parameter" and let users read the Globus Compute docs to understand the object they are creating. I think right now, users of this code are going to have to be understanding stuff about Globus Compute anyway.

I think it gives a better "lazy path" sustainability story.

Reply:
@benclifford I like the idea of passing in the `globus_compute_sdk.Executor` itself to the executor, but this would break the `init -> start` separation that `ParslExecutors` follow.

Reply:
We just discussed this, and the consensus is that passing along the executor is the most straightforward solution here.